addons/stock_account/models/product.py

```diff
@@ -705,7 +705,7 @@ def _compute_average_price(self, qty_invoiced, qty_to_invoice, stock_moves, is_r
         # price to estimate the anglo saxon price unit.
         missing = qty_to_invoice - qty_valued
         for sml in stock_moves.move_line_ids:
-            if not sml.owner_id or sml.owner_id == sml.company_id.partner_id:
+            if sml._should_exclude_for_valuation():
```

Comment:
Thank you for reducing both extracted methods to a single one :+1: 

In this place, however, we should negate the condition ;)
```suggestion
            if not sml._should_exclude_for_valuation():
```